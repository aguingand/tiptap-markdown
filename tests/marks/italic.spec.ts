import { expect, test } from "vitest";
import { parse, serialize } from "../utils";


test('parse markdown with "*"', () => {
    expect(parse('*example*')).toMatchSnapshot();
});
test('parse markdown with "_"', () => {
    expect(parse('_example_')).toMatchSnapshot();
})
test('parse html', () => {
    expect(parse('<em>example</em>')).toMatchSnapshot();
});
test('serialize', () => {
    expect(serialize('<em>example</em>')).toEqual('*example*\n');
    expect(serialize('<em data-markdown-marker="_">example</em>')).toEqual('_example_\n');
});
