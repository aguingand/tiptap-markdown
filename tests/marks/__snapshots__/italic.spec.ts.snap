// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parse html 1`] = `
[
  {
    "content": [
      {
        "marks": [
          {
            "attrs": {
              "data-markdown-marker": "*",
            },
            "type": "italic",
          },
        ],
        "text": "example",
        "type": "text",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`parse markdown with "*" 1`] = `
[
  {
    "content": [
      {
        "marks": [
          {
            "attrs": {
              "data-markdown-marker": "*",
            },
            "type": "italic",
          },
        ],
        "text": "example",
        "type": "text",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`parse markdown with "_" 1`] = `
[
  {
    "content": [
      {
        "marks": [
          {
            "attrs": {
              "data-markdown-marker": "_",
            },
            "type": "italic",
          },
        ],
        "text": "example",
        "type": "text",
      },
    ],
    "type": "paragraph",
  },
]
`;
